### PR TITLE
Noah MP updates for Prototype 8 (GFSv17)

### PR DESCRIPTION
**Pull request by HelinWei-NOAA**
This is the first part of [Noah MP upgrades for P8 ](https://github.com/NCAR/ccpp-physics/pull/729)- adding some extra surface fields to the model output. The other parts only involves CCPP-Physics.

Add some extra surface fields required to close the offline energy/water budget computation. These fields are listed below:
LAND FRACTION
INST PREC ADVEC HEAT
WATER IN AQUIFER
ACM EIWATER(CANOPY WATER EVAPORATION)
ACM PLANTTR( PLANT TRANSPIRATION)
ACM SOILSE (BARE SOIL EVAPORATION)
AVE PREC ADVEC HEAT

### Issue(s) addressed

Link the issues to be closed with this PR, whether in this repository, or in another repository.
(Remember, issues should always be created before starting work on a PR branch!)
- fixes #<issue_number>
- fixes noaa-emc/fv3atm/issues/<issue_number>

## Testing

How were these changes tested?  
What compilers / HPCs was it tested with?  
Are the changes covered by regression tests? (If not, why? Do new tests need to be added?)  
Have the ufs-weather-model regression test been run? On what platform?  
- Will the code updates change regression test baseline? If yes, why? Please show the baseline directory below.
- Please commit the regression test log files in your ufs-weather-model branch

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs
- waiting on noaa-emc/nems/pull/<pr_number>
- waiting on noaa-emc/fv3atm/pull/<pr_number>

### CODE REVIEW

File: ccpp/data/GFS_typedefs.F90 [hunk 1]
```diff
@@ -6415,6 +6425,7 @@ subroutine diag_create (Diag, IM, Model)
     allocate (Diag%snohfa  (IM))
     allocate (Diag%transa  (IM))
     allocate (Diag%sbsnoa  (IM))
+    allocate (Diag%paha    (IM))
```
Comment (grantfirl) [inline ccpp/data/GFS_typedefs.F90]: According to https://github.com/NCAR/ccpp-physics/pull/729, some of the additional variables are only used when NoahMP is active. You may want to conditionally allocate these when lsm==lsm_noahmp to save memory.

File: ccpp/data/GFS_typedefs.meta [hunk 2]
```diff
@@ -6375,6 +6375,13 @@
   dimensions = (horizontal_loop_extent)
   type = real
   kind = kind_phys
+[paha]
```
Comment (climbfuji) [inline ccpp/data/GFS_typedefs.meta:6378]: This array is now only allocated if the LSM is Noah MP. You need to add a line `active = (control_for_land_surface_scheme == identifier_for_noahmp_land_surface_scheme)` for each of those variables that are allocated conditionally. Please see the existing entry for variable `snowxy` for an example.

Comment (HelinWei-NOAA) [reply ccpp/data/GFS_typedefs.meta:6378]: made the modification

Comment (climbfuji) [reply ccpp/data/GFS_typedefs.meta:6378]: Thanks, that looks great!

**Review by climbfuji — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Pull request merged**